24H1KR3: Store Architectures (#1170)
vtexdocs/dev-portal-content: docs/guides/VTEX-Platform-Overview/store-architecture/store-architecture.md

---
title: "Store architecture"
slug: "store-architecture"
hidden: false
createdAt: "2024-05-23T10:18:55.338Z"
updatedAt: "2024-05-23T10:18:55.338Z"
excerpt: "Learn about the VTEX reference architectures"
seeAlso:
 - "/docs/guides/understanding-vtex-reference-architectures"
 - "/docs/guides/store-architecture-use-cases"
---

Store architecture is the main way to communicate a unified vision of a particular project to all stakeholders, ensuring clarity for everyone involved and translating business needs into software components and project requirements.

There are different types and levels of architecture, but as a general rule, it should work as a reference for both technical and business teams. Moreover, store architectures define and make it clear to all stakeholders what are the roles and responsibilities of each player, how different systems interoperate, and how data is transmitted between them.

> To enhance the understanding of the related aspects and show how VTEX connects with the ecosystem, we designed VTEX reference models of architecture, which are based on the more common digital commerce strategies. These reference architectures, also known as blueprints, are diagrams created to represent the interactions between VTEX modules and the business and are conceived as a starting point for a project. These reference architectures were designed following the guidelines which are in the [Understanding reference architectures](#LINK) guide.

Reviewer: Ficou faltando atualizar um link aqui. Acho que a doc ainda não existe, né?

> To enhance the understanding of the related aspects and show how VTEX connects with the ecosystem, we designed VTEX reference models of architecture, which are based on the more common digital commerce strategies. These reference architectures, also known as blueprints, are diagrams created to represent the interactions between VTEX modules and the business and are conceived as a starting point for a project. These reference architectures were designed following the guidelines which are in the [Understanding VTEX reference architectures](https://developers.vtex.com/docs/guides/understanding-vtex-reference-architectures) guide.

Reviewer 2: ```suggestion
To demonstrate how VTEX integrates with the ecosystem, we created VTEX reference architecture models based on common digital commerce strategies. These blueprints illustrate interactions between VTEX modules and the business, serving as a project starting point. Learn more about the structure of these models in [Understanding VTEX reference architectures](https://developers.vtex.com/docs/guides/understanding-vtex-reference-architectures).
```

> >ℹ️ To learn more about the main characteristics of the accounts that relate to the planning of your store’s architecture, check out the [Accounts and architecture](https://help.vtex.com/en/tracks/vtex-store-overview--eSDNk26pdvemF3XKM0nK9/4yPqZQyj0t675QpcG7H6yl) guide.

Reviewer 2: ```suggestion
>ℹ️ For details on account characteristics relevant to planning your store’s architecture, refer to the [Accounts and architecture](https://help.vtex.com/en/tracks/vtex-store-overview--eSDNk26pdvemF3XKM0nK9/4yPqZQyj0t675QpcG7H6yl) guide.
```

> Below are an overview of the main use cases for common scenarios:

Reviewer 2: ```suggestion
The main use cases for common scenarios are the following:
```

### Single account, multi-bindings

In this architecture, there is a single VTEX account bound to multiple languages and currencies, in a [multidomain](https://help.vtex.com/en/tracks/vtex-store-overview--eSDNk26pdvemF3XKM0nK9/4yPqZQyj0t675QpcG7H6yl#multistore) perspective.

![image](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@main/docs/guides/VTEX-Platform-Overview/store-architecture/multi-currency-language-single-account.png)

### Multi-account

In this architecture, there is a main VTEX account that will act as a seller in all secondary VTEX accounts, acting as [marketplaces](https://help.vtex.com/en/tutorial/estrategias-de-marketplace-na-vtex--tutorials_402#being-a-vtex-marketplace). These secondary accounts are conceived as [new environments](https://help.vtex.com/en/tracks/vtex-store-overview--eSDNk26pdvemF3XKM0nK9/4yPqZQyj0t675QpcG7H6yl#additional-environment).

![image](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@main/docs/guides/VTEX-Platform-Overview/store-architecture/multi-currency-language-multi-account.png)

> >ℹ️ To deep dive into these store architectures, check out the [Store architecture use cases](https://developers.vtex.com/docs/guides/store-architecture-use-cases) guide.

Reviewer 2: Reading this part now, I feel like this guide does not add much when compared to https://github.com/vtexdocs/dev-portal-content/pull/1171 _Store architecture use cases_.

- It feels repetitive to divide sections in the same way in both articles.
- I don't think _Store architecture use cases_ would get way too long if we just added a bit more information to the introduction.
- _Store architecture use cases_ seems stronger and more useful in terms of content and I don't think it should be "hidden" under this subcategory cover.

My suggestion is merging this introduction with the use cases guide, so for me the next steps would be:

1. Adjust content from lines 13 to 28 as suggested and then incorporate this part into the _Store architecture use cases_ guide introduction in https://github.com/vtexdocs/dev-portal-content/pull/1171. Make sure all images used in that guide are in that PR as well.
2. Close this PR.
3. Rename the _Store architecture use cases_ guide to _Store architecture_ and use it as the subcategory cover.
